main/streams/streams.c

```diff
@@ -1563,73 +1563,93 @@ PHPAPI zend_result _php_stream_copy_to_stream_ex(php_stream *src, php_stream *de
 			src->writepos == src->readpos) {
 		/* both php_stream instances are backed by a file descriptor, are not filtered and the
 		 * read buffer is empty: we can use copy_file_range() */
-		int src_fd, dest_fd, dest_open_flags = 0;
+		int src_fd, dest_fd, dest_open_flags;
 
-		/* get dest open flags to check if the stream is open in append mode */
-		php_stream_parse_fopen_modes(dest->mode, &dest_open_flags);
+		/* Some filesystems will cause failures if the max length is greater than the file length.
```

Comment:
Isn't this just a minor use case? Does it make sense to optimize for that and penalise (even though it might be minor penalisation but it's still extra syscall) majority of use cases?

Reply:
I thought the stat would be pretty cheap in comparison to the file IO, but you might be right about this. I'll think about it a bit more